### ISSUE TITLE
How to disable post request compression

### ISSUE THREAD

**Issue by deadash** [libcpr/cpr#683]
## Description
When I use a post request to upload a file, if the file is a bit big, it will be automatically compressed, and I need to upload it as it is.

## Expected Behavior
Upload without compression

## Actual Behavior
Will be compressed by gzip, the file header is pk

## Possible Fix
Don't know how to do it, maybe curl has the answer?

**Comment by COM8** [comment]
As far as I'm aware curl does not automatically compress files for upload. But perhaps it does it in case the server announces gzip support.

What happens if you do something like:
```c++
Session session;
session.SetUrl(url);
curl_easy_setopt(session.GetCurlHolder()->handle, CURLOPT_ACCEPT_ENCODING, "identity");
// [...] Your other code for the request
```

**Comment by deadash** [comment]
The client sends a POST request directly, how does it know whether the server supports it? Did the server reject it after sending it?

The following is to view the sent header
Accept-Encoding: deflate, gzip

Through your tips, after I modified the code
Accept-Encoding: identity

Lines of code related to this， https://github.com/libcpr/cpr/blob/bcb7729dd6d930820fb858f63c97ad559241b789/cpr/session.cpp#L749

**Comment by COM8** [comment]
Sorry, I do not understand your reply there.

I understood:
* This fixed it.
* We are enabling compression for send data with the line you are highlighting.

**Comment by deadash** [comment]
Yes, your answer fixed it.

I hope this setting can be easily set. Just like setting Proxies.

**Comment by leviliangtw** [comment]
I'd like to take this issue.

**Comment by barisconur** [comment]
I would like to work on this issue.

**Comment by KingKili** [comment]
@leviliangtw @barisconur, can you both agree on one of you? Not that you both create the same solution.

**Comment by leviliangtw** [comment]
@barisconur, have you started this issue already?
I think I have an idea for it.

**Comment by barisconur** [comment]
@leviliangtw I have not started yet. Do you have a chance to implement and test your idea?

**Comment by leviliangtw** [comment]
@barisconur Yes, let me get the chance. 
@KingKili I take this issue.

**Comment by leviliangtw** [comment]
@KingKili The PR [#746](https://github.com/libcpr/cpr/pull/746) is ready for this issue

**Comment by COM8** [comment]
Closing since this has been solved #746 .
